subprojects/core/src/main/java/org/gradle/cache/internal/GradleUserHomeCleanupService.java

```diff
@@ -20,13 +20,16 @@
 import org.gradle.internal.concurrent.Stoppable;
 import org.gradle.internal.logging.progress.ProgressLogger;
 import org.gradle.internal.logging.progress.ProgressLoggerFactory;
+import org.gradle.util.GUtil;
 
 import java.io.File;
+import java.util.Properties;
 
 public class GradleUserHomeCleanupService implements Stoppable {
 
     private static final long MAX_UNUSED_DAYS_FOR_RELEASES = 30;
     private static final long MAX_UNUSED_DAYS_FOR_SNAPSHOTS = 7;
+    private static final String DISABLE_CACHE_CLEANUP_PROPERTY = "org.gradle.cache.cleanup.disable";
```

Comment:
Please rename this property to `org.gradle.cache.cleanup` and change the logic to that `true` means enabled and `false` disabled.

Reply:
Should this also mean if the property does not exist that the cache deletion is disabled?

Reply:
No, its default value should be `true`.